examples/flax/summarization/run_summarization_flax.py

```diff
@@ -0,0 +1,800 @@
+            model_args.tokenizer_name, cache_dir=model_args.cache_dir, use_fast=model_args.use_fast_tokenizer
+        )
+    elif model_args.model_name_or_path:
+        tokenizer = AutoTokenizer.from_pretrained(
+            model_args.model_name_or_path, cache_dir=model_args.cache_dir, use_fast=model_args.use_fast_tokenizer
+        )
+    else:
+        raise ValueError(
+            "You are instantiating a new tokenizer from scratch. This is not supported by this script."
+            "You can do it from another script, save it, and load it from here, using --tokenizer_name."
+        )
+
+    if model_args.model_name_or_path:
+        model = FlaxAutoModelForSeq2SeqLM.from_pretrained(
+            model_args.model_name_or_path, config=config, seed=training_args.seed, dtype=getattr(jnp, model_args.dtype)
+        )
+    else:
+        model = FlaxAutoModelForSeq2SeqLM.from_config(
+            config, seed=training_args.seed, dtype=getattr(jnp, model_args.dtype)
+        )
+
+    if model.config.decoder_start_token_id is None:
+        raise ValueError("Make sure that `config.decoder_start_token_id` is correctly defined")
+
+    prefix = data_args.source_prefix if data_args.source_prefix is not None else ""
+
+    # Preprocessing the datasets.
+    # We need to tokenize inputs and targets.
+    if training_args.do_train:
+        column_names = dataset["train"].column_names
+    elif training_args.do_eval:
+        column_names = dataset["validation"].column_names
+    elif training_args.do_predict:
+        column_names = dataset["test"].column_names
+    else:
+        logger.info("There is nothing to do. Please pass `do_train`, `do_eval` and/or `do_predict`.")
+        return
+
+    # Get the column names for input/target.
+    dataset_columns = summarization_name_mapping.get(data_args.dataset_name, None)
+    if data_args.text_column is None:
+        text_column = dataset_columns[0] if dataset_columns is not None else column_names[0]
+    else:
+        text_column = data_args.text_column
+        if text_column not in column_names:
+            raise ValueError(
+                f"--text_column' value '{data_args.text_column}' needs to be one of: {', '.join(column_names)}"
+            )
+    if data_args.summary_column is None:
+        summary_column = dataset_columns[1] if dataset_columns is not None else column_names[1]
+    else:
+        summary_column = data_args.summary_column
+        if summary_column not in column_names:
+            raise ValueError(
+                f"--summary_column' value '{data_args.summary_column}' needs to be one of: {', '.join(column_names)}"
+            )
+
+    # Temporarily set max_target_length for training.
+    max_target_length = data_args.max_target_length
+
+    # In Flax every Seq2Seq model should define this, as the models don't accept labels
+    # we need to prepare the decoder_input_ids here
+    prepare_decoder_input_ids_fn = model.prepare_decoder_input_ids_from_labels
```

Comment:
As each model has a different way of preparing `decoder_input_ids`, think we should add `prepare_decoder_input_ids_from_labels` to every seq2seq flax model

Reply:
I don't have strong opinion about this but If we do this the users will always need to import `shift_tokens_right` function during training with
`from transformers.models.modeling_fl... import shift_tokens_right`

which seems a bit awkward, as for most of our models for training we just need to import model and tokenizer from the main init. So having a method or static method on the model instance would be nice IMO.

Reply:
IMO Flax is definitely different from PyTorch in a sense that it is not enough to just import the model and tokenizer for training. Given that in Flax the gradients are computed by calling `jax.grad` on the loss function, we just cannot have the same API as in PyTorch. 

Given that IMO, the training script should be responsible for everything training related and not the model. So I do prefer to not attach `shift_tokens_right` to the model. 

Regarding the solution proposed above: Yes the user will have to import a non-public method `shift_tokens_right` from the model, but I don't see a problem with that. 

Another solution would actually be to just not important the method and rewrite it in the training script. As far as I remember it's only MBart that uses a different `shift_tokens_right` method. 

What do you think?

Reply:
Yes, makes sense! Will adapt the script accordingly